arduino-ide-extension/src/common/protocol/sketches-service.ts

```diff
@@ -151,6 +153,89 @@ export interface Sketch extends SketchRef {
   readonly rootFolderFileUris: string[]; // `RootFolderFiles` (does not include the main sketch file)
 }
 export namespace Sketch {
+  // (non-API) exported for the tests
+  export const defaultSketchFolderName = 'sketch';
+  // (non-API) exported for the tests
+  export const defaultFallbackFirstChar = '0';
+  // (non-API) exported for the tests
+  export const defaultFallbackChar = '_';
+  // (non-API) exported for the tests
+  export const invalidSketchFolderNameMessage = nls.localize(
+    'arduino/sketch/invalidSketchName',
+    'The name must start with a letter or number, followed by letters, numbers, dashes, dots and underscores. Maximum length is 63 characters.'
+  );
+  const invalidCloudSketchFolderNameMessage = nls.localize(
+    'arduino/sketch/invalidCloudSketchName',
+    'The name must consist of basic letters, numbers or underscores. Maximum length is 36 characters.'
+  );
+  /**
+   * `undefined` if the candidate sketch folder name is valid. Otherwise, the validation error message.
+   * Based on the [specs](https://arduino.github.io/arduino-cli/latest/sketch-specification/#sketch-folders-and-files).
+   */
+  export function validateSketchFolderName(
+    candidate: string
+  ): string | undefined {
+    return /^[0-9a-zA-Z]{1}[0-9a-zA-Z_\.-]{0,62}$/.test(candidate)
+      ? undefined
+      : invalidSketchFolderNameMessage;
+  }
+
+  /**
+   * `undefined` if the candidate cloud sketch folder name is valid. Otherwise, the validation error message.
+   * Based on how https://create.arduino.cc/editor/ works.
+   */
+  export function validateCloudSketchFolderName(
+    candidate: string
+  ): string | undefined {
+    return /^[0-9a-zA-Z_]{1,36}$/.test(candidate)
```

Comment:
I still have the same request as https://github.com/arduino/arduino-ide/pull/1821#pullrequestreview-1268619776:

>We need to get a confirmation that it really is necessary to disallow the - and . characters in Cloud sketches. That seems a completely unnecessary departure from the Arduino Sketch Specification to me.

My impression was that the decision was made that `-` and `.` can be allowed since there is no technical reason to disallow it and no work required in any other project to allow it.

Reply:
9687fc6